Using Metadata("struct:field:name", "-") in combination with Tag() causes generation failure

## Expected Behavior

I would expect using `Metadata("struct:field:name", "-")` to hide an attribute from the resulting HTTP response body, but not from anything that needs it internally, such as `Tag()` processing.

## Actual Behavior

What actually seems to happen is that it hides the field from even being generated which causes generation to make invalid code that does not compile and fails.

## Steps to Reproduce the Problem (including complete and simple design if relevant)

Given a design:

```golang
package thingy

import (
    . "goa.design/goa/http/design"
    . "goa.design/goa/http/dsl"
)

var Thingys = Type("Thingys", func(){
    Attribute("etag", String)
    Attribute("not_modified", String, func() {
        Enum("true", "false")
        Metadata("struct:field:name", "-")
    })
})

var _ = API("Thingy", func() {
})

var _ = Service("Thingy", func() {
    Method("index", func() {
        Payload(func() {
            Attribute("etag", String)
        })
        Result(Thingys)

        HTTP(func() {
            GET("/")
            Body(Empty)
            Header("etag:If-None-Match")
            Response(StatusOK, func() {
                Header("etag:ETag")
            })
            Response(StatusNotModified, func() {
                Tag("not_modified", "true")
                Body(Empty)
            })
        })
    })
})
```

`goa gen ...` errors with:

```
exit status 1
C:\cygwin64\home\dthrift\go\src\github.com\douglaswth\thingy\gen\http\thingy\client\types.go:37:11: expected selector or type assertion, found '!='
C:\cygwin64\home\dthrift\go\src\github.com\douglaswth\thingy\gen\http\thingy\client\types.go:38:13: expected selector or type assertion, found '=='
C:\cygwin64\home\dthrift\go\src\github.com\douglaswth\thingy\gen\http\thingy\client\types.go:43:3: expected '}', found 'EOF'

========
Content:
// Code generated with goa v2.0.0-wip, DO NOT EDIT.
//
// Thingy HTTP client types
//
// Command:
// $ goa gen github.com/douglaswth/thingy

package client

import (
        "unicode/utf8"
        thingy "github.com/douglaswth/thingy/gen/thingy"
        goa "goa.design/goa"
)

// IndexOKResponseBody is the type of the "Thingy" service "index" endpoint
// HTTP response body.
type IndexOKResponseBody struct {
         *string `form:"not_modified,omitempty" json:"not_modified,omitempty" xml:"not_modified,omitempty"`
}
// NewIndexThingysNotModified builds a "Thingy" service "index" endpoint result
// from a HTTP "NotModified" response.
func NewIndexThingysNotModified() *thingy.Thingys {
                        return &thingy.Thingys{
                        }
        }
// NewIndexThingysOK builds a "Thingy" service "index" endpoint result from a
// HTTP "OK" response.
func NewIndexThingysOK(body *IndexOKResponseBody, etag *string, ) *thingy.Thingys {
                v := &thingy.Thingys{
NotModified: body.NotModified,
}
                        v.Etag = etag
                return v}
// Validate runs the validations defined on IndexOKResponseBody
func (body *IndexOKResponseBody) Validate() (err error) {
        if body. != nil {
if !(*body. == "true" || *body. == "false") {
        err = goa.MergeErrors(err, goa.InvalidEnumValueError("body.not_modified", *body., []interface{}{"true", "false"}))
}
}
        return
}

```

## Specifications

  - goa Version: v2
  - Platform (Linux/Windows etc.): Windows

--- a/codegen/goify.go
+++ b/codegen/goify.go
@@ -13,6 +13,11 @@ import (
 // firstUpper is true the first character of the identifier is uppercase
 // otherwise it's lowercase.
 func Goify(str string, firstUpper bool) string {
+	// Optimize trivial case
+	if str == "" {
+		return ""
+	}
+
 	// Remove optional suffix that defines corresponding transport specific
 	// name.
 	idx := strings.Index(str, ":")
@@ -24,6 +29,14 @@ func Goify(str string, firstUpper bool) string {
 
 	// remove trailing invalid identifiers (makes code below simpler)
 	runes = removeTrailingInvalid(runes)
+
+	// If all characters are invalid produce a default value.
+	if len(runes) == 0 {
+		if firstUpper {
+			return "Val"
+		}
+		return "val"
+	}
 
 	w, i := 0, 0 // index of start of word, scan
 	for i+1 <= len(runes) {

--- a/codegen/types_test.go
+++ b/codegen/types_test.go
@@ -73,6 +73,7 @@ func TestGoify(t *testing.T) {
 		firstUpper bool
 		expected   string
 	}{
+		"empty":                                             {"", false, ""},
 		"first upper false":                                 {"blue_id", false, "blueID"},
 		"first upper false normal identifier all lower":     {"blue", false, "blue"},
 		"first upper false and UUID":                        {"blue_uuid", false, "blueUUID"},
@@ -86,7 +87,7 @@ func TestGoify(t *testing.T) {
 		"only UUID and firstupper false":                    {"UUID", false, "uuid"},
 		"consecutives invalid identifiers firstupper false": {"[[fields___type]]", false, "fieldsType"},
 		"consecutives invalid identifiers":                  {"[[fields___type]]", true, "FieldsType"},
-		"invalid identifiers":                               {"[[", false, ""},
+		"invalid identifiers":                               {"[[", false, "val"},
 		"middle upper firstupper false":                     {"MiddleUpper", false, "middleUpper"},
 		"middle upper":                                      {"MiddleUpper", true, "MiddleUpper"},
 	}

